### PR TITLE
feat(auth): Add domain to api scope, remove from scope group.

### PR DESCRIPTION
**Pull request by valurefugl**
https://www.notion.so/Add-domain-to-API-Scope-table-a5f47e8dce234313b824fdb0edb6d074

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/auth-api-lib/src/lib/resources/models/api-scope-group.model.ts [hunk 1]
```diff
@@ -67,16 +64,6 @@ export class ApiScopeGroup extends Model {
...
 
-  @Column({
-    type: DataType.STRING,
-    allowNull: false,
-  })
-  @ApiProperty({
-    example: '@island.is',
-  })
-  @ForeignKey(() => Domain)
-  domainName!: string
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/resources/models/api-scope-group.model.ts:78]: The api scope group needs also to belong to just a single domain.

File: libs/auth-api-lib/src/lib/resources/resources.service.ts [hunk 2]
```diff
@@ -966,7 +966,7 @@ export class ResourcesService {
           offset: offset,
           where: { name: { [Op.iLike]: `%${searchString}%` } },
           order: [['name', 'asc']],
-          include: [ApiScopeGroup],
+          include: [ApiScope],
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/resources/resources.service.ts:969]: We should just return the domains without including the scopes or groups.

**Review by saevarma — CHANGES_REQUESTED** [review]

File: libs/auth-api-lib/src/lib/resources/models/api-scope-group.model.ts (hunk 1 above)
Comment (valurefugl) [reply libs/auth-api-lib/src/lib/resources/models/api-scope-group.model.ts]: Fixed.

File: libs/auth-api-lib/src/lib/resources/resources.service.ts (hunk 2 above)
Comment (valurefugl) [reply libs/auth-api-lib/src/lib/resources/resources.service.ts]: FIxed.

File: libs/auth-api-lib/src/lib/resources/resources.service.ts [hunk 3]
```diff
@@ -879,28 +893,32 @@ export class ResourcesService {
...
+    return this.apiScopeGroupModel.create({ id: id, ...group })
   }
 
   /** Updates an existing ApiScopeGroup */
   async updateApiScopeGroup(
     group: ApiScopeGroupDTO,
     id: string,
   ): Promise<[number, ApiScopeGroup[]]> {
-    return this.apiScopeGroup.update(
+    if (await this.isDomainDifferentFromScopes(id, group)) {
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/resources/resources.service.ts:904]: Same here, rename it `assertSameAsScopes` and throw there.

**Review by saevarma — CHANGES_REQUESTED** [review]

File: libs/auth-api-lib/migrations/20221010090424-add-domain-to-api-scope.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        { transaction },
+      )
+
+      await queryInterface.changeColumn(
+        'api_scope',
+        'domain_name',
+        {
+          type: Sequelize.STRING,
+          allowNull: false,
+          defaultValue: '@island.is',
```
[the remark this answers is not in the document]
Comment (valurefugl) [reply libs/auth-api-lib/migrations/20221010090424-add-domain-to-api-scope.js]: Removed default value.

File: libs/auth-api-lib/src/lib/resources/models/api-scope.model.ts [hunk 5]
```diff
@@ -94,6 +96,15 @@ export class ApiScope extends Model<ModelAttributes, CreationAttributes> {
   @ApiProperty()
   description!: string
 
+  @Column({
+    type: DataType.STRING,
+    allowNull: false,
+    defaultValue: '@island.is',
```
[the remark this answers is not in the document]
Comment (valurefugl) [reply libs/auth-api-lib/src/lib/resources/models/api-scope.model.ts]: Removed default value.

File: libs/auth-api-lib/src/lib/resources/resources.service.ts [hunk 6]
```diff
@@ -569,6 +570,15 @@ export class ResourcesService {
   async createApiScope(apiScope: ApiScopesDTO): Promise<ApiScope> {
     this.logger.debug('Creating a new api scope')
 
+    if (apiScope.groupId) {
+      const scopeGroup = await this.apiScopeGroupModel.findByPk(
+        apiScope.groupId,
+      )
```
[the remark this answers is not in the document]
Comment (valurefugl) [reply libs/auth-api-lib/src/lib/resources/resources.service.ts]: Yikes, fixed.

File: libs/auth-api-lib/src/lib/resources/resources.service.ts [hunk 7]
```diff
@@ -569,6 +570,15 @@ export class ResourcesService {
...
+      if (await this.isDomainDifferentFromGroup(apiScope)) {
+        throw new BadRequestException('Scope domain must match group domain.')
+      }
+    }
```
[the remark this answers is not in the document]
Comment (valurefugl) [reply libs/auth-api-lib/src/lib/resources/resources.service.ts]: Fixed.

File: libs/auth-api-lib/src/lib/resources/resources.service.ts (hunk 3 above)
Comment (valurefugl) [reply libs/auth-api-lib/src/lib/resources/resources.service.ts]: Fixed.

File: libs/auth-api-lib/migrations/20221010090424-add-domain-to-api-scope.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        { transaction },
+      )
+
+      await queryInterface.sequelize.query(
+        `UPDATE api_scope SET domain_name = (SELECT domain_name FROM api_scope_group WHERE id = api_scope.group_id);`,
+        { transaction },
+      )
+
+      await queryInterface.sequelize.query(
+        `UPDATE api_scope SET domain_name = '@island.is' where domain_name IS NULL;`,
```
[the remark this answers is not in the document]
Comment (saevarma) [reply libs/auth-api-lib/migrations/20221010090424-add-domain-to-api-scope.js]: We should just set set those to `@island.is`

**Review by saevarma — APPROVED** [review]
Nice job 🙌

**Pull request merged**